libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/AsyncSelectComponents.tsx

```diff
@@ -0,0 +1,119 @@
+// (C) 2021 GoodData Corporation
+import React from "react";
+import {
+    components as ReactSelectComponents,
+    InputProps,
+    MenuListComponentProps,
+    OptionProps,
+} from "react-select";
+import { Typography } from "../../../Typography";
+import { LoadingMask } from "../../../LoadingMask";
+import {
+    ISelectErrorOption,
+    ISelectOption,
+    isGranteeItem,
+    isGranteeUser,
+    isSelectErrorOption,
+} from "./types";
+import { getGranteeItemTestId } from "./utils";
+
+export const EmptyRenderer = (): JSX.Element => {
+    return null;
+};
+
+export const LoadingMessageRenderer = (): JSX.Element => {
+    return (
+        <div className="gd-share-dialog-loading-mask-container">
+            <LoadingMask size="small" />
+        </div>
+    );
+};
+
+// missing export type for NoOptionsMessage from react-select
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const NoOptionsMessageRenderer = (props: any): JSX.Element => {
+    return (
+        <div className="s-gd-share-dialog-no-option">
+            <ReactSelectComponents.NoOptionsMessage {...props} />
+        </div>
+    );
+};
+
+export const MenuListRendered = (props: MenuListComponentProps<ISelectOption, false>): JSX.Element => {
+    return (
+        <div className="s-gd-share-dialog-menu">
+            <ReactSelectComponents.MenuList {...props} />
+        </div>
+    );
+};
+
+export const InputRendered = (props: InputProps): JSX.Element => {
+    return (
+        <div className="gd-share-dialog-input s-gd-share-dialog-input">
+            <ReactSelectComponents.Input {...props} />
+        </div>
+    );
+};
+
+const OptionContentRenderer = (item: ISelectOption): JSX.Element => {
+    if (isGranteeUser(item.value)) {
+        return (
+            <>
+                {item.value.name} <span className={"option-email"}>{item.value.email}</span>
+            </>
+        );
+    }
+
+    return <> {item.label} </>;
+};
+
+export const ErrorOptionRenderer = (errorOption: ISelectErrorOption): JSX.Element => {
+    return (
+        <div className={`gd-share-dialog-option-error s-gd-share-dialog-option-error`}>
+            <span className="gd-share-dialog-option-error-content">{errorOption.label}</span>
+        </div>
+    );
+};
+
+export const OptionRenderer = (props: OptionProps<ISelectOption, false>): JSX.Element => {
+    const { className, cx, isFocused, innerRef, innerProps, data } = props;
+
+    if (isSelectErrorOption(data)) {
+        return ErrorOptionRenderer(data);
+    }
+
+    let sTestStyle = "";
+
+    if (isGranteeItem(data.value)) {
+        sTestStyle = getGranteeItemTestId(data.value, "option");
```

Comment:
I would change this to ternary operator to save lots of lines and remove `let`.

Reply:
You can also use `cx` possibly as `cx({[getGranteeItemTestId(data.value, "option");]: isGranteeItem(data.value)})`

Reply:
done

Reply:
We let is as it is isGranteeItem is type guard and we need render s-style just for options that has data type of GranteeItem